sublime_music/adapters/adapter_base.py

```diff
@@ -667,6 +674,17 @@ def scrobble_song(self, song: Song):
         """
         raise self._check_can_error("scrobble_song")
 
+    def set_song_rating(self, song_id: str, rating: int):
+        """
+        Rate the given song
+
+        Calls the endpoint described in http://www.subsonic.org/pages/api.jsp#setRating
+
+        :param song_id: A string which uniquely identifies the file (song)
+        :param rating: The rating between 1 and 5 (inclusive), or 0 to remove the rating
```

Comment:
Adapters should be generic (eventually I might actually get around to adding other adapters for other services).

I think we should not make `0` a magic number, instead explicitly pass `None` so that the caching layer works better. The subsonic adapter can translate that to a 0 for the actual API call.

```suggestion
        Rate the given song.

        :param song_id: The ID of the song to rate.
        :param rating: The rating between 1 and 5 (inclusive), or `None` to remove the rating.
```

Reply:
None is used now rather than 0.